src/freertos_drivers/common/MCP2515Can.hxx

```diff
@@ -570,7 +577,21 @@ private:
         ::write(spi, &rts, 1);
     }
 
-    unsigned txPending; /**< transmission in flight */
+    void request_io_operation()
+    {
+        if (!ioPending)
+        {
+            ioPending = true;
+            sem.post();
+        }
+    }
+
+    unsigned txPending : 2; /**< transmission in flight */
+    unsigned gpoData   : 2; /**< local copy of the I/O expansion output data */
+    unsigned gpiData   : 3; /**< local copy of the I/O expansion input data */
+    unsigned ioPending : 1; /**< true if an I/O update is pending */
+    unsigned reserved  : 24; /**< unused bits */
```

Comment:
consider removing this line. It doesn't help (alignment is automatic in C) but can cause unnecessary memory usage if someone adds just one bit later without adjusting the count here.

Reply:
fixed